From david-b@pacbell.net  Tue Jul 28 10:34:49 2009
From: Nicolas Ferre <nicolas.ferre@atmel.com>
Date: Mon, 27 Jul 2009 14:59:24 -0700
Subject: USB: at91: modify OHCI driver to allow shared interrupts
To: Greg KH <greg@kroah.com>
Cc: Nicolas Ferre <nicolas.ferre@rfo.atmel.com>
Message-ID: <200907271459.24520.david-b@pacbell.net>
Content-Disposition: inline


From: Nicolas Ferre <nicolas.ferre@atmel.com>

At91sam9g45 series has a set of high speed USB interfaces.
The host driver is an EHCI with its companion OHCI. OHCI is
always handled by ohci-at91.c.
This wrapper is just modified to allow IRQ sharing
between two controllers.

Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-at91.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/ohci-at91.c
+++ b/drivers/usb/host/ohci-at91.c
@@ -148,7 +148,7 @@ static int usb_hcd_at91_probe(const stru
 	at91_start_hc(pdev);
 	ohci_hcd_init(hcd_to_ohci(hcd));
 
-	retval = usb_add_hcd(hcd, pdev->resource[1].start, IRQF_DISABLED);
+	retval = usb_add_hcd(hcd, pdev->resource[1].start, IRQF_SHARED);
 	if (retval == 0)
 		return retval;
 
